### ISSUE TITLE
[Bug] No more new reviews the next day after upgrade to 2.15.0beta3

### ISSUE THREAD

**Issue by szpak** [ankidroid/Anki-Android#8846]
###### Reproduction Steps

1. Upgrade to 2.15.0beta3 (here from the previous beta - but not 2.15.x - or stable version if no beta was available)
2. Do some daily review
3. Wait until next day

###### Expected Result

I see the new reviews on the following day.

###### Actual Result

There are no new reviews. Only leftovers from the preview day were propagated. It is somehow problematic to continue study :-/.

It could also affect new cards, however, I haven't it configured for any of my decks.

Btw, I don't know how much it is reproducible. If problematic, I can play with parallel build and some other account.

###### Questions to the team

1. What would you recommend to bring my new reviews back (I already synchronized with the online account the previous day)? Downgrade to the  previous version? Resync the online account with the desktop version and try to get updates there?

###### Debug info

```
AnkiDroid Version = 2.15.0beta3
Android Version = 11
Manufacturer = Google
Model = Pixel 3a
Hardware = sargo
Webview User Agent = Mozilla/5.0 (Linux; Android 11; Pixel 3a Build/[redacted]; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/90.0.4430.210 Mobile Safari/537.36
ACRA UUID = [redacted]
Scheduler = std2
Crash Reports Enabled = true
DatabaseV2 Enabled = true
```

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
This is *probably* the v2 scheduler new timezone code taking effect. I would *guess* that it is a one-day realignment and should be normal next day, or you might be able to alter things via various settings of when next day starts

possibly related #8652  ?

**Comment by szpak** [comment]
Might be. I have new day start set to 4am and I was - accidentally - reviewing some of the cards several minutes after 4am (something common with the aforementioned report). I will check it the following day.

Btw, I'm in [CEST](https://time.is/CEST) right now (UTC+2).

**Comment by szpak** [comment]
Two updates.
1. Might be not relevant, but at 4:20am I had the counter of "studied today" reset to zero, but old the old reviews (from the previous day) were shown.
2. When I started a review in one of the decks where I still had of reviews during the day (N+1), but at the later hour (here 6pm) all "expected" reviews turn up when I answered the first review. Therefore, maybe the problem is just with refreshing the visible state (especially after the original trigger doesn't kick up on time due to suggested time zone issues)?

Btw, thanks @mikehardy for your analysis.

**Comment by mikehardy** [comment]
Interesting - I would bet then, that tomorrow things "look normal" *if* you are not reviewing across the new-day time boundary.

However, there may be something to what you mention: how is the app handling things exactly as the new-day time boundary is crossed? We may have some visual artifacts there that pre-new-day and some that are post - I've never investigated that - I tried to set my new day to a time when I am never ever going to be reviewing Anki cards, and I was successful ;)

**Comment by szpak** [comment]
> I tried to set my new day to a time when I am never ever going to be reviewing Anki cards, and I was successful ;)

Well, in fact I reviewed at 4am for the first time ever ;-), so there is a chance that that problem was there in the past. However, I have some (very) old parallel build also installed on that phone and I remember, I checked that the "studied today" counter was not reset at that time (although, I haven't done any reviews at 4am). So maybe these "time zone things" might be important here

**Comment by szpak** [comment]
I was able to reproduce that a few days later (checking at 4am first and later opening - probably only minimized app in the morning. However, the integration with the app refresh the status, so it's not a big issue.

**Comment by mikehardy** [comment]
I believe this was fixed in 2.15.x subsequent releases - I'm going to close this as there was definitely a scheduling issue in 2.15.0 fixed quickly in 2.15.1, but we can always reopen if it is still happening
